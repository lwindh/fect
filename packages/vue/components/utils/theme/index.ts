export * from './color'
export * from './prop-types'
