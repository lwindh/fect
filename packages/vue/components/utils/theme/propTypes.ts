const tuple = <T extends string[]>(...args: T): T => {
  const noStr = [...args].map((_) => typeof _ !== 'string').includes(true)
  if (noStr) {
    throw new Error('[Fect] type Error. type is not string!')
  }
  return args
}

const normalSizes = tuple('mini', 'small', 'medium', 'large')

const normalTypes = tuple('default', 'success', 'warning', 'error')

const themeTypes = tuple('dark', 'light')

const justifyTypes = tuple(
  'start',
  'end',
  'center',
  'space-around',
  'space-between',
)

const alignTypes = tuple('top', 'middle', 'bottom')

const snippetCopyTypes = tuple('default', 'silent', 'prevent')

const snippetStyleTypes = tuple(
  'default',
  'secondary',
  'success',
  'warning',
  'error',
  'lite',
  'dark',
)

const placeTypes = tuple('topLeft', 'topRight', 'bottomLeft', 'bottomRight')

const loadingTypes = tuple('default', 'cube', 'wave')

export {
  tuple,
  normalSizes,
  normalTypes,
  themeTypes,
  justifyTypes,
  alignTypes,
  snippetCopyTypes,
  snippetStyleTypes,
  placeTypes,
  loadingTypes,
}

export type NormalSizes = typeof normalSizes[number]
export type NormalTypes = typeof normalTypes[number]
export type ButtonTypes = NormalTypes | 'secondary'
export type ThemeTypes = typeof themeTypes[number]
export type JustifyTypes = typeof justifyTypes[number]
export type AlignTypes = typeof alignTypes[number]
export type SnippetCopyTypes = typeof snippetCopyTypes[number]
export type SnippetStyleTypes = typeof snippetStyleTypes[number]
export type PlaceTypes = typeof placeTypes[number]
export type LoadingTypes = typeof loadingTypes[number]
