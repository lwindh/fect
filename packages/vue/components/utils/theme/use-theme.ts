// theme provide light and dark

import { useState } from '../useState'
import { watch } from 'vue'
export type Theme = 'light-theme' | 'dark-theme'

export const useTheme = () => {
  const [theme, setTheme] = useState<Theme>(localStorage.getItem('theme') || 'light-theme')

  const themeChange = () => setTheme(theme.value === 'light-theme' ? 'dark-theme' : 'light-theme')

  watch(
    theme,
    (pre) => {
      localStorage.setItem('theme', pre)
      if (typeof document === 'undefined') return
      const root = document.querySelector('html')
      root?.setAttribute('class', theme.value)
    },
    { immediate: true }
  )

  return {
    themeChange,
    theme,
  }
}
