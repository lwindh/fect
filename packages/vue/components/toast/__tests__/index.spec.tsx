import Toast from '../toast'
import { NormalTypes } from '../../utils/theme/propTypes'
import { mount } from '@vue/test-utils'
describe('Toast', () => {
  it('should render as element', () => {
    const wrapper = mount(Toast, {
      props: {
        text: 'Test Message',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-toast__message').text()).toBe('Test Message')
  })

  it('should support different types', () => {
    const wrapper = mount({
      setup() {
        return {
          types: ['default', 'success', 'warning', 'error'] as NormalTypes[],
        }
      },
      render() {
        return (
          <>
            {this.types.map((type, idx) => (
              <Toast type={type} key={`${type}+${idx}`} text={idx} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
