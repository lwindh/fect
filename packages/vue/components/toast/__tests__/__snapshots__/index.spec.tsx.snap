// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render as element 1`] = `
<div class="fect-toast  " style="--toast-translate: translate3d(0, calc(100% + -75px + --20px), --1px) scale(1.05); --toast-opacity: 1; --toast-shadow: var(--fect-shadowSmall);">
  <div class="fect-toast__message">Test Message</div>
  <!---->
</div>
`;

exports[`Toast should support different types 1`] = `
<div class="fect-toast  " style="--toast-translate: translate3d(0, calc(100% + -75px + --20px), --1px) scale(1.05); --toast-opacity: 1; --toast-shadow: var(--fect-shadowSmall);">
  <div class="fect-toast__message">0</div>
  <!---->
</div>
<div class="fect-toast  " style="--toast-translate: translate3d(0, calc(100% + -75px + --20px), --1px) scale(1.05); --toast-opacity: 1; --toast-shadow: var(--fect-shadowSmall); color: white;">
  <div class="fect-toast__message">1</div>
  <!---->
</div>
<div class="fect-toast  " style="--toast-translate: translate3d(0, calc(100% + -75px + --20px), --1px) scale(1.05); --toast-opacity: 1; --toast-shadow: var(--fect-shadowSmall); color: white;">
  <div class="fect-toast__message">2</div>
  <!---->
</div>
<div class="fect-toast  " style="--toast-translate: translate3d(0, calc(100% + -75px + --20px), --1px) scale(1.05); --toast-opacity: 1; --toast-shadow: var(--fect-shadowSmall); color: white;">
  <div class="fect-toast__message">3</div>
  <!---->
</div>
`;
